nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/AbstractScriptProcessor.java

```diff
@@ -216,7 +220,8 @@ public void setup() {
      *
      * @see org.apache.nifi.processors.script.ScriptEngineConfigurator
      */
-    protected void setupEngine() {
+    protected void setupEngines(int numberOfScriptEngines) {
+        engineQ = new LinkedBlockingQueue<>(numberOfScriptEngines);
         ClassLoader originalContextClassLoader = Thread.currentThread().getContextClassLoader();
         try {
             ProcessorLog log = getLogger();
```

Comment:
GitHub won't let me put the line comment on 229, but that can throw an NPE if the `scriptEngineName` is not defined when this method is run. In normal execution, that won't occur, but during test configuration, the `setup` method can be called before the `context` is defined and provided.